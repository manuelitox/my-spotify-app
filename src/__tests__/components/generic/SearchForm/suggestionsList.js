import React from 'react'
import { Link } from 'react-router-dom'
import { shallow } from 'enzyme'

import SuggestionsList from 'components/generic/SearchForm/suggestionsList'

describe('Component | SuggestionsList', () => {
  const SUGGESTIONS = [
    { id: '1', name: 'Alicia Keys', images: [] },
    { id: '2', name: 'Queen', images: [] },
    { id: '3', name: 'John Mayer', images: [] },
    { id: '4', name: 'ACDC', images: [] },
    { id: '5', name: 'Blink 182', images: [] },
    { id: '6', name: 'Bon Jovi', images: [] },
    { id: '7', name: '2pac', images: [] }          
  ]
  let component 
  beforeEach( () => {
    component = shallow(<SuggestionsList isOpen data={ SUGGESTIONS } numberOfArtists={ 4 } closeSuggestions={ jest.fn() } />)
  })
  it('should render a component', () => {
    expect(component).toMatchSnapshot()
  })
  it('should display the `To` parameter of the first Link', () => {
    // replace the spaces ' ' to '-' into the Artist name.
    const to = component.find(Link).at(0).props().to
    expect(to).toEqual('/artist/Alicia-Keys/1')
  })
  it('should render null because isOpen is false', () => {
    component = shallow(<SuggestionsList isOpen={ false } closeSuggestions={ jest.fn() } />)    
    expect(component).toMatchSnapshot()
  })
  it('should render null because numberOfArtists is 0', () => {
    component = shallow(<SuggestionsList isOpen numberOfArtists={ 0 } closeSuggestions={ jest.fn() } />)    
    expect(component).toMatchSnapshot()
  })
})