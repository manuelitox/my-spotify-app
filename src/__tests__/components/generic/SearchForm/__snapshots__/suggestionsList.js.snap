// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | SuggestionsList should render a component 1`] = `
<styled.ul>
  <li
    key="1"
  >
    <Link
      onClick={[MockFunction]}
      replace={false}
      to={
        Object {
          "pathname": "/artist/Alicia-Keys/1",
          "search": "?#access_token=access_token123123",
        }
      }
    >
      <img
        alt="Alicia Keys"
        height="40"
        src="placeholderArtist.svg"
        width="40"
      />
      Alicia Keys
    </Link>
  </li>
  <li
    key="2"
  >
    <Link
      onClick={[MockFunction]}
      replace={false}
      to={
        Object {
          "pathname": "/artist/Queen/2",
          "search": "?#access_token=access_token123123",
        }
      }
    >
      <img
        alt="Queen"
        height="40"
        src="placeholderArtist.svg"
        width="40"
      />
      Queen
    </Link>
  </li>
  <li
    key="3"
  >
    <Link
      onClick={[MockFunction]}
      replace={false}
      to={
        Object {
          "pathname": "/artist/John-Mayer/3",
          "search": "?#access_token=access_token123123",
        }
      }
    >
      <img
        alt="John Mayer"
        height="40"
        src="placeholderArtist.svg"
        width="40"
      />
      John Mayer
    </Link>
  </li>
  <li
    key="4"
  >
    <Link
      onClick={[MockFunction]}
      replace={false}
      to={
        Object {
          "pathname": "/artist/ACDC/4",
          "search": "?#access_token=access_token123123",
        }
      }
    >
      <img
        alt="ACDC"
        height="40"
        src="placeholderArtist.svg"
        width="40"
      />
      ACDC
    </Link>
  </li>
  <li
    key="5"
  >
    <Link
      onClick={[MockFunction]}
      replace={false}
      to={
        Object {
          "pathname": "/artist/Blink-182/5",
          "search": "?#access_token=access_token123123",
        }
      }
    >
      <img
        alt="Blink 182"
        height="40"
        src="placeholderArtist.svg"
        width="40"
      />
      Blink 182
    </Link>
  </li>
  <li
    key="6"
  >
    <Link
      onClick={[MockFunction]}
      replace={false}
      to={
        Object {
          "pathname": "/artist/Bon-Jovi/6",
          "search": "?#access_token=access_token123123",
        }
      }
    >
      <img
        alt="Bon Jovi"
        height="40"
        src="placeholderArtist.svg"
        width="40"
      />
      Bon Jovi
    </Link>
  </li>
  <li
    key="7"
  >
    <Link
      onClick={[MockFunction]}
      replace={false}
      to={
        Object {
          "pathname": "/artist/2pac/7",
          "search": "?#access_token=access_token123123",
        }
      }
    >
      <img
        alt="2pac"
        height="40"
        src="placeholderArtist.svg"
        width="40"
      />
      2pac
    </Link>
  </li>
</styled.ul>
`;

exports[`Component | SuggestionsList should render null because isOpen is false 1`] = `""`;

exports[`Component | SuggestionsList should render null because numberOfArtists is 0 1`] = `""`;
